lib/DDG/Spice/BikeSharing/CitiBikeNYC.pm

```diff
@@ -10,6 +10,12 @@ spice wrap_jsonp_callback => 1;
 spice is_cached => 0;
 spice proxy_cache_valid => '200 304 15m';
 
+spice alt_to => {
+	status => {
```

Comment:
@marianosimone the `alt_to` path prepends the name of the subdirectory, if one exists:

`/js/spice/subdir_name/alt_to/`

In this case, it's inside `BikeSharing` so the path is `/js/spice/bike_sharing/status`.

If there is no subdirectory, then the path is just `/js/spice/alt_to` -- so in those cases the `alt_to` name should be named to indicate what IA it belongs to.

The KXCD IA is a good example to follow. We have `lib/DDG/Spice/XKCD/Display.pm` which defines an `alt_to`, called "status".

Thus the endpoints are `/js/spice/xkcd/display` and `/js/spice/xkcd/latest` and the JS calls the latter: https://github.com/duckduckgo/zeroclickinfo-spice/blob/master/share/spice/xkcd/display/xkcd_display.js#L44

----

> Removing citi_bike_nyc obviously doesn't work, because we are one dir below that. But the inner URL rewriting system doesn't seem to understand nested files, returning the same for both /js/spice/bike_sharing/citi_bike_nyc/status and (the non-existent) /js/spice/bike_sharing/citi_bike_nyc/fuuuuu; in both cases, it returns the first level reponse

What's happening here is that you're calling the endpoint for the CitiBikeNyc.pm `spice to` and sending it the param "status". However, your `spice to` doesn't use `$1` anywhere, so the `status` word is ignored. 

AFAIK, we can't nest endpoints within each other because of how the NGINX rewrites work, meaning any extension of `/js/spice/bike_sharing/citi_bike_nyc/` is going to be re-written.

If we try to have them nested, we'd run into issues with reserved keywords:

i.e. `/js/spice/sub_dir/endpoint_a/keyword1/keyword2`, `/js/spice/sub_dir/endpoint_a/endpoint_a_b/keyword1/keyword2`

How do you differentiate the 2nd url from `/js/spice/sub_dir/endpoint_a/keyword1/keyword2/keyword3`?

NGINX just applied a regular expression to the string `i.e. `keyword1/keyword2/keyword3`, attempts to capture part of it, and proxy the request to the `spice to` URL, replacing `$1, $2, $3` if they're defined and captured.

Does that make more sense?